src/p2p/net_node.inl

```diff
@@ -2507,7 +2532,7 @@ namespace nodetool
       return 1;
     }
 
-    if(has_too_many_connections(context.m_remote_address))
+    if(has_too_many_connections(preferred_na))
```

Comment:
The check must be from the external address. Otherwise a node could connect multiple times, using this new field as way to bypass this check entirely.